File: python/bubble_sort.py
def bubble_sort(l):

  for i in range(len(l)):
    swap = False

    for j in range(len(l) - i - 1):
      if l[j] > l[j+1]:
        l[j], l[j+1] = l[j+1], l[j]
        swap = True

    if swap is False:
      break
    
  return l

File: js/bubbleSort.js
const bubbleSort = function (list) {
  for (let i = 0; i < list.length; i++) {
    swap = false;

    for (let j = 0; j < list.length - i - 1; j++) {
      if (list[j] > list[j + 1]) {
        temp = list[j];
        list[j] = list[j + 1];
        list[j + 1] = temp;
        swap = true;
      }
    }

    if (swap === false) break;
  }
  return list;
};

module.exports = { bubbleSort };
